### PR TITLE
Minor refactoring of `cedar_policy_validator::SchemaError`

### PR DESCRIPTION
**Pull request by shaobo-he-aws**
## Checklist for requesting a review

The change in this PR is (choose one, and delete the other options):

- [ ] A breaking change requiring a major version bump to `cedar-policy` (e.g., changes to the signature of an existing API).
- [ ] A backwards-compatible change requiring a minor version bump to `cedar-policy` (e.g., addition of a new API).
- [ ] A bug fix or other functionality change requiring a patch to `cedar-policy`.
- [x] A change "invisible" to users (e.g., documentation, changes to "internal" crates like `cedar-policy-core`, `cedar-validator`, etc.)
- [ ] A change (breaking or otherwise) that only impacts unreleased or experimental code.

I confirm that this PR (choose one, and delete the other options):

- [ ] Updates the "Unreleased" section of the CHANGELOG with a description of my change (required for major/minor version bumps).
- [x] Does not update the CHANGELOG because my change does not significantly impact released code.

I confirm that [`cedar-spec`](https://github.com/cedar-policy/cedar-spec) (choose one, and delete the other options):

- [x] Does not require updates because my change does not impact the Cedar Dafny model or DRT infrastructure.
- [ ] Requires updates, and I have made / will make these updates myself. (Please include in your description a timeline or link to the relevant PR in `cedar-spec`, and how you have tested that your updates are correct.)
- [ ] Requires updates, but I do not plan to make them in the near future. (Make sure that your changes are hidden behind a feature flag to mark them as experimental.)
- [ ] I'm not sure how my change impacts `cedar-spec`. (Post your PR anyways, and we'll discuss in the comments.)

## Disclaimer

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: cedar-policy/src/api.rs [hunk 1]
```diff
@@ -867,29 +872,24 @@ impl From<cedar_policy_validator::SchemaError> for SchemaError {
             cedar_policy_validator::SchemaError::CycleInActionHierarchy => {
                 Self::CycleInActionHierarchy
             }
-            cedar_policy_validator::SchemaError::EntityTypeParseError(e) => {
-                Self::EntityTypeParse(e)
-            }
-            cedar_policy_validator::SchemaError::NamespaceParseError(e) => Self::NamespaceParse(e),
-            cedar_policy_validator::SchemaError::CommonTypeParseError(e) => {
+            cedar_policy_validator::SchemaError::ParseEntityType(e) => Self::EntityTypeParse(e),
+            cedar_policy_validator::SchemaError::ParseNamespace(e) => Self::NamespaceParse(e),
```
Comment (khieta) [inline cedar-policy/src/api.rs:876]: Looks a little weird here that the names are reversed for the `cedar-policy` errors vs. the `cedar-policy-validator` errors. I prefer the names with `Parse` first.

File: cedar-policy-validator/src/err.rs [hunk 2]
```diff
@@ -146,15 +141,22 @@ impl std::fmt::Display for ContextOrShape {
...
+    ActionAttributes(Vec<String>),
 }
 
 impl std::fmt::Display for UnsupportedFeature {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
         match self {
-            UnsupportedFeature::OpenRecordsAndEntities => write!(
+            Self::OpenRecordsAndEntities => write!(
                 f,
                 "Records and entities with additional attributes are not yet implemented."
```
Comment (khieta) [inline cedar-policy-validator/src/err.rs:153]:
```suggestion
                "records and entities with additional attributes are not yet implemented"
```

File: cedar-policy-validator/src/err.rs [hunk 3]
```diff
@@ -146,15 +141,22 @@ impl std::fmt::Display for ContextOrShape {
...
             ),
+            Self::ActionAttributes(attrs) => write!(
+                f,
+                "Action declared with attributes: [{}]",
```
Comment (khieta) [inline cedar-policy-validator/src/err.rs:157]:
```suggestion
                "action declared with attributes: [{}]",
```

**Review by khieta — APPROVED** [review]

File: cedar-policy/src/api.rs (hunk 1 above)
Comment (shaobo-he-aws) [reply cedar-policy/src/api.rs:876]: This PR is restricted to `cedar_policy_validator`. I'll update the one in the Cedar APIs in #231.

**Review by khieta — APPROVED** [review]

**Review by cdisselkoen — APPROVED** [review]

**Pull request merged**